### PR TITLE
Fix workspace registry implementation

### PR DESCRIPTION
**Pull request by KattMingMing**
I realized there were a couple issues with the previous implementation so I cleaned it up here. I have another branch that modifies the quick open with this implementation that works very well 👍 

Also fixes vscode tslint issues for Hygiene 

Want to get this in to unblock @rothfels

### CODE REVIEW

File: src/vs/platform/workspace/common/workspace.ts [hunk 1]
```diff
@@ -78,6 +82,13 @@ export interface IWorkspaceRevState {
 	zapRef?: string;
 }
 
+declare class Map<K, V> {
```
Comment (rothfels) [inline src/vs/platform/workspace/common/workspace.ts:85]: Why do we have to do this? What are the commented out lines for?

File: src/vs/platform/workspace/common/workspace.ts [hunk 2]
```diff
@@ -89,7 +100,8 @@ export class WorkspaceContextService implements IWorkspaceContextService {
 	constructor(workspace: IWorkspace) {
 		this.workspace = workspace;
 		this.workspaceEmitter = new Emitter<IWorkspace>();
-		this.workspaceRegistry.set(this.workspace.resource.toString(), this.workspace);
+		const workspaceRegistryKey = workspace.resource.with({ fragment: '', query: '' }).toString();
```
Comment (rothfels) [inline src/vs/platform/workspace/common/workspace.ts:101]: We don't want to be doing this in the long run, please make a TODO that when I change the URI scheme on Sourcegraph we no longer need to do `.with({ fragment: '', query: '' })`. Why do you need to here (just curious)?

File: src/vs/platform/workspace/common/workspace.ts [hunk 3]
```diff
@@ -120,17 +132,18 @@ export class WorkspaceContextService implements IWorkspaceContextService {
 		return null;
 	}
 
-	public tryGetWorkspaceFromRegistry(resource: URI): IWorkspaceRevState | undefined {
+	public tryGetWorkspaceFromRegistry(resource: URI): IWorkspace | undefined {
 		return this.workspaceRegistry.get(resource.toString());
```
Comment (rothfels) [inline src/vs/platform/workspace/common/workspace.ts]: Doesn't this need to do the same `.with({ fragment: '', query: '' })` as above/below?

**Review by rothfels — APPROVED** [review]

File: src/vs/platform/workspace/common/workspace.ts (hunk 1 above)
Comment (KattMingMing) [reply src/vs/platform/workspace/common/workspace.ts:85]: Apparently `Map` is not defined

Comment (KattMingMing) [reply src/vs/platform/workspace/common/workspace.ts:85]: Also the previously imported `LinkedMap` does not update when `set(` is called, it returns false if a value was already present.

**Pull request merged**